crates/rpc/src/module.rs

```diff
@@ -119,6 +123,46 @@ impl Module {
 
         Ok(self)
     }
+
+    pub fn register_subscription<Subscription, WSAnySubscriptionEvent: 'static + Send + Clone>(
+        mut self,
+        subscription_name: &'static str,
+        subscription_answer_name: &'static str,
+        unsubscription_name: &'static str,
+        subscription: Subscription,
+        ws_broadcast_tx: RPCSender<WSAnySubscriptionEvent>,
+    ) -> anyhow::Result<Self>
+    where
+        Subscription: (Fn(
+                RpcContext,
+                SubscriptionSink,
+                &RPCSender<WSAnySubscriptionEvent>,
+            ) -> SubscriptionResult)
+            + Copy
+            + Send
+            + Sync
+            + 'static,
+    {
+        use anyhow::Context;
+        use jsonrpsee::types::Params;
+
+        metrics::register_counter!("rpc_subscription_calls_total", "subscription" => subscription_name);
+
+        let subscription_callback = move |_params: Params<'_>, sink, context: Arc<RpcContext>| {
+            subscription((*context).clone(), sink, &ws_broadcast_tx.clone())
```

Comment:
If i understand the semantics of new counter correctly it should then be incremented here.

Reply:
Aye, committed your suggestion :)